src/Talis/Persona/Client/OAuthClients.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Talis\Persona\Client;
+
+class OAuthClients extends Base
+{
+    /**
+     * Update a users OAuth client
+     * @param string $clientId
+     * @param array $properties
+     * @param string $token
+     * @return mixed
+     * @access public
+     * @throws \InvalidArgumentException
+     * @throws \Exception
+     */
+    public function updateOAuthClient($clientId, $properties, $token)
+    {
+        if(!is_string($clientId) || trim($clientId) === '')
+        {
+            throw new \InvalidArgumentException('Invalid guid');
+        }
+        if(!is_array($properties) || empty($properties))
+        {
+            throw new \InvalidArgumentException('Invalid properties');
+        }
+        if(!is_string($token) || trim($token) === '')
+        {
+            throw new \InvalidArgumentException('Invalid token');
+        }
+
+        $url = $this->config['persona_host'].'/clients/'.$clientId;
+
+        try
+        {
+            $client = $this->personaPatchOAuthClient($url, $properties, $token);
+            return $client;
+        } catch(\Exception $e)
+        {
+            throw new \Exception('OAuth client not updated');
+        }
+    }
+
+    /**
+     *
+     * @param string $url
+     * @param array $query
+     * @param string $token
+     * @return mixed
+     * @access protected
+     * @throws \Exception
+     */
+    protected function personaPatchOAuthClient($url, $query, $token)
```

Comment:
should this be $query? i.e. is it named correctly...isn't this the oauth client data to patch?